src/app/Plans/components/Wizard/MappingForm.tsx

```diff
@@ -0,0 +1,143 @@
+import * as React from 'react';
+import {
+  Checkbox,
+  Grid,
+  GridItem,
+  Radio,
+  TextInput,
+  Title,
+  Form,
+  FormGroup,
+  Flex,
+  FlexItem,
+} from '@patternfly/react-core';
+import spacing from '@patternfly/react-styles/css/utilities/Spacing/spacing';
+
+import SimpleSelect, { OptionWithValue } from '@app/common/components/SimpleSelect';
+import { IStorageMapping, MappingType, INetworkMapping } from '@app/Mappings/types';
+import { MappingBuilder, IMappingBuilderItem } from '@app/Mappings/components/MappingBuilder';
+import { MappingSource, MappingTarget } from '@app/Mappings/types';
+
+interface IMappingFormProps {
+  mappingType: MappingType;
+  mappingList: IStorageMapping[] | INetworkMapping[];
+  availableSources?: MappingSource[];
+  availableTargets?: MappingTarget[];
+}
+
+const MappingForm: React.FunctionComponent<IMappingFormProps> = ({
+  mappingType,
+  mappingList,
+  availableSources = [],
+  availableTargets = [],
+}: IMappingFormProps) => {
+  const mappingOptions = Object.values(mappingList).map((mapping) => ({
+    toString: () => mapping.name,
+    value: mapping,
+  })) as OptionWithValue<IStorageMapping | INetworkMapping>[];
+
+  const [isCreateMappingChecked, toggleCreateMapping] = React.useReducer(
+    (isCreateMappingChecked) => !isCreateMappingChecked,
+    false
+  );
+
+  const [isSaveNewMapping, toggleSaveNewMapping] = React.useReducer(
+    (isSaveNewMapping) => !isSaveNewMapping,
+    false
+  );
+
+  const [mapping, setMapping] = React.useState<IStorageMapping | INetworkMapping | null>(null);
```

Comment:
There's a `Mapping` union type that covers storage and network, this could be `<Mapping | null>`, not really important though.